packages/ember-metal/lib/array.js

```diff
@@ -101,7 +101,7 @@ var filter = defineNativeShim(ArrayPrototype.filter, function(fn, context) {
   var length = this.length;
 
   for (i = 0; i < length; i++) {
-    if (this.hasOwnProperty(i)) {
+    if (typeof this.hasOwnProperty !== 'function' || this.hasOwnProperty(i)) {
```

Comment:
The cross browser builds are finicky, please change this back as Stef mentioned and I'll restart the build if it fails again.